src/reducers/index.js

```diff
@@ -0,0 +1,291 @@
+                    editMovie: {
+                        ...state.modals.editMovie,
+                        isOpen: true
+                    }
+
+                }
+            }
+        case HIDE_MODAL_EDIT_MOVIE :
+            return {
+                ...state,
+                modals: {
+                    ...state.modals,
+                    editMovie: {
+                        ...state.modals.editMovie,
+                        isOpen: false
+                    }
+
+                }
+            }
+        case SHOW_MODAL_DELETE_MOVIE :
+            return {
+                ...state,
+                modals: {
+                    ...state.modals,
+                    deleteMovie: {
+                        ...state.modals.deleteMovie,
+                        isOpen: true
+                    }
+
+                }
+            }
+        case HIDE_MODAL_DELETE_MOVIE :
+            return {
+                ...state,
+                modals: {
+                    ...state.modals,
+                    deleteMovie: {
+                        ...state.modals.deleteMovie,
+                        isOpen: false
+                    }
+
+                }
+            }
+        case HANDLE_ADD_MOVIE_FORM_FIELD: {
+            const target = action.payload;
+            console.log(target.value);
+            return {
+                ...state,
+                forms: {
+                    ...state.forms,
+                    movieForm: {
+                        ...state.forms.movieForm,
+                        [target.id]: {
+                            ...state.forms.movieForm[target.id],
+                            value: target.value,
+                        },
+                    }
+                }
+            }
+        }
+        case HANDLE_GENRE_CHANGE: {
+            return {
+                ...state,
+                forms: {
+                    ...state.forms,
+                    movieForm: {
+                        ...state.forms.movieForm,
+                        genres: {
+                            ...state.forms.movieForm.genres,
+                            value: action.payload
+                        }
+                    }
+                }
+            }
+        }
+        case SET_EDIT_MOVIE_ID: {
+            const id = action.payload;
+            return {
+                ...state,
+                editMovieId: id
+            }
+        }
+        case REMOVE_EDIT_MOVIE_ID: {
+            return {
+                ...state,
+                editMovieId: null
+            }
+        }
+        case SET_DELETE_MOVIE_ID: {
+            const id = action.payload;
+            return {
+                ...state,
+                deleteMovieId: id
+            }
+        }
+        case REMOVE_DELETE_MOVIE_ID: {
+            return {
+                ...state,
+                deleteMovieId: null
+            }
+        }
+        default:
+            return state;
+    }
+}
```

Comment:
To fat reducer. You shouldn't use 1 fat reducer just split it to logical part of application and manipulate separatly.
https://redux.js.org/api/combinereducers
```js
const movieReducer = initialState => ({
    // ..code
});

const filterReducer = initialState => ({
    // ...code
});

const modalReducer = initialState => ({
    // code
});

const reducer = combineReducer({
    movies: movieReducer,
    filter: filterReducer,
    modal: modalReducer
});
```